### ISSUE TITLE
Error in TemporalDateTime fromJson parser

### ISSUE THREAD

**Issue by martinzuccotti** [aws-amplify/amplify-flutter#579]
**Describe the bug**
The parser has an error in the logic for the decimal part. When it is applied to a datetime with precision less than one second, the result obtained is wrong.

**To Reproduce**
Try to parse any iso8601String with decimal part (less than 9 digits) included and compare it with the result. 

**Details**
The related test is this
```dart
test(''AWSDateTime from string YYYY-MM-DDThh:mm:ss.sssZ success'', () async {
    TemporalDateTime time =
        TemporalDateTime.fromString("1995-05-03T03:30:25.999999999Z");

    expect(time.getOffset(), Duration());
    expect(time.getDateTimeInUtc(),
        DateTime.utc(1995, 05, 03, 03, 30, 25, 999, 999));
    expect(time.format(), "1995-05-03T03:30:25.999999999Z");
  });
```
It's ok, but it isn't complete. Only works with date times of 9 digit decimal part. 

A complementary test might be this (note that it now fails)
```dart
test('AWSDateTime from string YYYY-MM-DDThh:mm:ss.sssZ success', () async {
    TemporalDateTime time =
        TemporalDateTime.fromString("1995-05-03T03:30:25.999Z");

    expect(time.getOffset(), Duration());
    expect(time.getDateTimeInUtc(),
        DateTime.utc(1995, 05, 03, 03, 30, 25, 999, 0));
    expect(time.format(), "1995-05-03T03:30:25.999000000Z");
  });
```

**Comment by martinzuccotti** [comment]
A possible solution is using the padRigh function in the line 102 of packages/amplify_datastore_plugin_interface/lib/src/types/temporal/temporal_datetime.dart

It would look like this

```dart
TemporalDateTime.fromString(String iso8601String) {
    ...
    DateTime dateTime = DateTime.parse(match.group(1).split(".")[0]);

    int totalNanoseconds = Temporal.getIntOr0(match.group(4).padRight(9, "0"));
    int milliseconds = totalNanoseconds ~/ 1000000;
    int microseconds = (totalNanoseconds ~/ 1000) % 1000;
    _nanoseconds = totalNanoseconds % 1000;
    ...
}
```

**Comment by b-cancel** [comment]
This is a bug report I just filed that includes a couple of bugs I found and the workarounds to each
after reading your bug report I think it could help
https://github.com/aws-amplify/amplify-flutter/issues/822

**Comment by dnys1** [comment]
This was released with version 0.2.7
